### PR TITLE
feat: add sorting to solutions/pagination

### LINKED ISSUE #1058: [Feature]: Add sorting to solutions/pagination
### Description

Can reference the comment section for the feature

- [ ] as the app grows well need to paginate (unsure if it does already tbh)
- [ ] sort by upvotes
- [ ] sort by comments
- [ ] sort by new/old

### Explanation

_No response_

### PR DESCRIPTION
**Pull request by mechakin**
## Description
Added pagination to solutions and sort by upvotes, comments, and by newest/oldest solution.

## Related Issue
Closes #1058

## How Has This Been Tested?
Tested locally on Mac.

### CODE REVIEW

File: apps/web/src/app/[locale]/challenge/_components/comments/index.tsx
Comment (bautistaaa) [inline apps/web/src/app/[locale]/challenge/_components/comments/index.tsx:234]: thank you for doing this. was on my todos

**Comment by mechakin** [comment]
forgot to do the prettier check lol

**Comment by bautistaaa** [comment]
the sorting appears to be bugged. tested on this url: https://typehero-git-fork-mechakin-1058-add-pagina-b87b4a-trash-company.vercel.app/challenge/hello-world/solutions

i also noticed if i switch to a different solution and sort i briefly see the previous cache of a split second. cache keys might need to be tweaked

**Comment by mechakin** [comment]
got it, i'll fix the cache and the sorting. looks like it's just the most replies sort that's bugged

**Comment by bautistaaa** [comment]
> got it, i'll fix the cache and the sorting. looks like it's just the most replies sort that's bugged

i think oldest comments didnt work either but unsure if that filter is even useful. can you delete it?

**Comment by mechakin** [comment]
yeah i don't see much use for it either, i can delete it

**Comment by bautistaaa** [comment]
going to hold off merging https://github.com/typehero/typehero/pull/1075 until this is merged.

@aadamw can you update the conflicts after? i can help here if youre busy nbd (unsure if theres much work anyways)

**Comment by aadamw** [comment]
@bautistaaa no problem dude

**Comment by mechakin** [comment]
@bautistaaa @aadamw the sorting and the cache should be fixed now

**Comment by bautistaaa** [comment]
most replies is not sorted correctly. the UI also disappears after sorting a couple times. seems random but maybe the state is somehow being cleared.
CleanShot 2023-11-10 at 11 24 36@2x
CleanShot 2023-11-10 at 11 25 55@2x

**Comment by mechakin** [comment]
i think the url <https://typehero-git-fork-mechakin-1058-add-pagina-b87b4a-trash-company.vercel.app/challenge/hello-world/solutions> isn't properly updating the changes i've made. locally the sorting works for me:
Screenshot 2023-11-10 at 3 40 11 PM

the reason the ui disappears is because the old prisma query counting the replies was bugged which caused no data to be returned. that resulted in the state being cleared once it has been updated. i fixed the query with the new commits so i'm not sure why they're not showing up in the url

**Comment by bautistaaa** [comment]
> i think the url https://typehero-git-fork-mechakin-1058-add-pagina-b87b4a-trash-company.vercel.app/challenge/hello-world/solutions isn't properly updating the changes i've made. locally the sorting works for me: Screenshot 2023-11-10 at 3 40 11 PM
> 
> the reason the ui disappears is because the old prisma query counting the replies was bugged which caused no data to be returned. that resulted in the state being cleared once it has been updated. i fixed the query with the new commits so i'm not sure why they're not showing up in the url

ah shoot okay let me look locally

**Comment by bautistaaa** [comment]
looks good locally!

**Review by bautistaaa — APPROVED** [review]

**Pull request merged**
